site/components/form/Select.tsx

```diff
@@ -67,15 +66,8 @@ const Select = <T extends object>({
                         id={`${inputId}-input`}
                         defaultValue={useDefaultValue ? value ?? "" : undefined}
                         value={!useDefaultValue ? value ?? "" : undefined}
-                        onBlur={
-                            !updateOnChange
-                                ? (e) => handleBlur(e.target.value, inputName, stateUpdater, setErrors, schema)
-                                : undefined
-                        }
                         onChange={
-                            updateOnChange
-                                ? (e) => handleBlur(e.target.value, inputName, stateUpdater, setErrors, schema)
-                                : undefined
+                            updateOnChange ? (e) => handleChange(e.target.value, inputName, stateUpdater) : undefined
```

Comment:
We could use stateUpdater function instead of handleChange as this function invokes the stateUpdater without any  other changes.

Reply:
done